matches/dao/match-dao.go

```diff
@@ -0,0 +1,176 @@
+package dao
+
+import (
+	"database/sql"
+	"errors"
+
+	// pq acts as the driver for SQL requests
+	_ "github.com/lib/pq"
+)
+
+const connStr = "user=postgres dbname=postgres host=matches-db sslmode=disable"
+
+// MatchGetResponse contains the information stored about a given match
+type MatchGetResponse struct {
+	ID        int
+	UserOne   int
+	UserTwo   int
+	MatchedOn string
+}
+
+// MatchListResponse contains the information stored about all matches
+type MatchListResponse struct {
+	IDs []int
+}
+
+// MatchCreateRequest contains the information required to create a new match
+type MatchCreateRequest struct {
+	UserOne int
+	UserTwo int
```

Comment:
This allows me to provide no request body and still create a match - since it assumes the default values of 0 for each integer. Should this be allowed? 

```
❯❯❯ curl -X POST localhost:8000/api/match -d '{}'
{"ID":3,"UserOne":0,"UserTwo":0,"MatchedOn":"2020-02-12T14:36:21.62474Z"}
```

Reply:
I asked the same thing in the Slack chat, I think not, sounds like you agree

Reply:
Assuming that `SERIAL` is used as the key type in the `User` table, it can never be 0 (https://www.postgresqltutorial.com/postgresql-serial/) - so we could validate it has to be > 1?

Reply:
I think if the field isn't passed explicitly in the body then it shouldn't change.

Reply:
> I think if the field isn't passed explicitly in the body then it shouldn't change.

There must be a nice way of doing this, I'll look into it in the tidy-up.